npm.js

```diff
@@ -63,23 +65,13 @@ function clone(a) {
   return b;
 }
 
-// avoid storing passwords as plain text in config
-function encodeCredentials(auth) {
-  return new Buffer(encodeURIComponent(auth.username) + ':' + encodeURIComponent(auth.password)).toString('base64');
-}
-function decodeCredentials(str) {
-  var auth = new Buffer(str, 'base64').toString('ascii').split(':');
-  return {
-    username: decodeURIComponent(auth[0]),
-    password: decodeURIComponent(auth[1])
-  };
-}
-
 var NPMLocation = function(options, ui) {
+  var npmrc = new Npmrc();
+
   this.ui = ui;
   this.name = options.name;
   // default needed during upgrade time period
-  this.registryURL = options.registry || defaultRegistry;
+  this.registryURL = options.registry || npmrc.getRegistry() || defaultRegistry;
```

Comment:
Note that this will only work properly given https://github.com/jspm/jspm-cli/issues/758.